src/common/types/GSNStatistics.ts

```diff
@@ -0,0 +1,125 @@
+import { Address, IntString, ObjectMap, SemVerString } from './Aliases'
+import {
+  DepositedEventInfo,
+  GNSContractsEvent, HubAuthorizedEventInfo, HubUnauthorizedEventInfo,
+  RelayRegisteredEventInfo,
+  StakeAddedEventInfo, StakeInfo, StakePenalizedEventInfo,
+  StakeUnlockedEventInfo, StakeWithdrawnEventInfo, TransactionRejectedByPaymasterEventInfo, TransactionRelayedEventInfo
+} from './GSNContractsDataTypes'
+import PingResponse from '../PingResponse'
+import { GSNContractsDeployment } from '../GSNContractsDeployment'
+import { EventData } from 'web3-eth-contract'
+
+export interface PingResult {
+  pingResponse?: PingResponse
+  error?: Error
+}
+
+/**
+ * Simplified structure derived from 'EventData' for better types support and user-visible output.
+ */
+export interface EventTransactionInfo<T extends GNSContractsEvent> {
+  eventData: EventData
+  // TODO: is it useful? There are untyped 'returnValues' in eventData anyways
+  returnValues: T
+}
+
+export enum RelayServerStakeStatus {
+  /** only staked, but never registered on currently selected RelayHub */
+  STAKE_LOCKED,
+  /** stake unlocked but not yet withdrawn */
+  STAKE_UNLOCKED,
+  /** stake withdrawn */
+  STAKE_WITHDRAWN,
+  /** stake has been penalized */
+  STAKE_PENALIZED
+}
+
+export interface RelayHubEvents {
+  depositedEvents?: Array<EventTransactionInfo<DepositedEventInfo>>
+  relayRegisteredEvents: Array<EventTransactionInfo<RelayRegisteredEventInfo>>
+  transactionRelayedEvents: Array<EventTransactionInfo<TransactionRelayedEventInfo>>
+  transactionRejectedEvents: Array<EventTransactionInfo<TransactionRejectedByPaymasterEventInfo>>
+}
+
+export interface StakeMangerEvents {
+  allEvents: EventData[]
+  stakeAddedEvents: Array<EventTransactionInfo<StakeAddedEventInfo>>
+  stakeUnlockedEvents: Array<EventTransactionInfo<StakeUnlockedEventInfo>>
+  stakeWithdrawnEvents: Array<EventTransactionInfo<StakeWithdrawnEventInfo>>
+  stakePenalizedEvents: Array<EventTransactionInfo<StakePenalizedEventInfo>>
+  hubAuthorizedEvents: Array<EventTransactionInfo<HubAuthorizedEventInfo>>
+  hubUnauthorizedEvents: Array<EventTransactionInfo<HubUnauthorizedEventInfo>>
+}
+
+export interface RelayServerRegistrationInfo {
+  lastRegisteredUrl: string
+  lastRegisteredBaseFee: IntString
+  lastRegisteredPctFee: IntString
+  pingResult: PingResult
```

Comment:
ping is the only field that is "last" ... url, fee, pct are truly the last value, but are much less temporary than the ping result..
I think the "last" prefix should not be used, as it doesn't add anything, just confusing.

Reply:
Will remove